### ISSUE TITLE
linuxserver/unifi-controller do not start 

### ISSUE THREAD

**Issue by zaggash** [linuxserver/docker-unifi-controller#24]
I'm starting the container on :latest
It takes ages to start.

Am I the only one with this behaviour ?
Every restart of the container take about 3 minutes.

Looks like the S6 processes are way too long to work with I do not know what.

```
# docker exec -ti unifi_controller.1.v9ra1agu8vinkmw6nrzzagntg ps ax
  PID TTY      STAT   TIME COMMAND
    1 ?        Ss     0:00 s6-svscan -t0 /var/run/s6/services
   30 ?        S      0:00 foreground  if   /etc/s6/init/init-stage2-redirfd   f
   31 ?        S      0:00 s6-supervise s6-fdholderd
   42 ?        S      0:00 if  /etc/s6/init/init-stage2-redirfd  foreground   if
   43 ?        S      0:00 foreground  if   if    s6-echo    -n    --    [s6-ini
   49 ?        S      0:00 if  if  -t   s6-test   -d   /var/run/s6/etc/cont-init
  178 ?        S      0:00 if  pipeline   s6-ls   -0   --   /var/run/s6/etc/cont
  181 ?        S      0:00 forstdin -o 0 -0 -- i importas -u i i if  s6-echo  --
  182 ?        Z      0:00 [s6-ls] <defunct>
  183 ?        Z      0:00 [s6-sort] <defunct>
  204 ?        S      0:00 foreground  /var/run/s6/etc/cont-init.d/20-config  im
  206 ?        S      0:00 bash /var/run/s6/etc/cont-init.d/20-config
  215 ?        D      0:00 chown -R abc:abc /config /usr/lib/unifi
  221 pts/0    Rs+    0:00 ps ax
```

Didn't have time to dig into it yet.

**Comment by aptalca** [comment]
Is it the chown that takes too long?

**Comment by zaggash** [comment]
Not sure.
When I `docker stats <container id>` the blockIO are growing and the memory too.
Up to the point where it stops and the processes start

**Comment by aptalca** [comment]
Mine starts in 2 seconds.

Can you see if it's only happening for newly created containers or restarted ones as well? The chown can cause similar issues due to an overlay2 bug in a fresh container where the perms are being changed inside the container

**Comment by zaggash** [comment]
I'll be interrested to see the overlay2 bug.
I'm using others images with chown inside a container withtout any issue tho.

**Comment by zaggash** [comment]
I just see that copaed to the other images, this one chown the whole  /usr/lib/unifi/ which contains symlinks to /config.
this folder have about 5k files.
Maybe it is the culprit here.

**Comment by aptalca** [comment]
5k files? That's definitely it.

Here's the bug report: https://github.com/docker/for-linux/issues/388

**Comment by zaggash** [comment]
Did a bit more testing here and found a workaround.

I updated to kernel 5.0 from 18.04 LTS Ubunto version => Nothing changed.
I changed my graph driver to aufs and now it works correctly.
So the issue is definitly because of overlay2 and it is yet not fixed on kernel 5.0.

**Comment by aptalca** [comment]
Yeah, it's a really annoying bug since overlay2 is the default storage driver.

For heimdall (around 15k files), the workaround we use is downloading the tar blob during build and we extract it during first run, it takes a split second that way compared to 10 mins of chown.

I'll see if we can do that here
